src/main/java/com/sessions/session9/homeworkOOP2/User.java

```diff
@@ -0,0 +1,158 @@
+    {
+        Random random = new Random();
+
+        return random.nextInt(origin, bound);
+    }
+
+    public User()
+    {
+        age = autoGeneratedAge(18, 80);
+        name = "Unkown";
+        dateOfBirth = LocalDate.now();
+        nationality = Nationality.Unknown;
+        hasJob = false;
+        idUser = incrementIDs++;
+        nrOfUsers++;
+    }
+
+    public User(final int age, final String name, final LocalDate dateOfBirth, final Nationality nationality, final boolean hasJob)
+    {
+        this.age = age;
+        this.name = name;
+        this.dateOfBirth = dateOfBirth;
+        this.nationality = nationality;
+        this.hasJob = hasJob;
+        this.idUser = incrementIDs++;
+        nrOfUsers++;
+    }
+
+    public int getAge()
+    {
+        return age;
+    }
+
+    public String getName()
+    {
+        return name;
+    }
+
+    public LocalDate getDateOfBirth()
+    {
+        return dateOfBirth;
+    }
+
+    public Nationality getNationality()
+    {
+        return nationality;
+    }
+
+    public boolean isHasJob()
+    {
+        return hasJob;
+    }
+
+    public int getIdUser()
+    {
+        return idUser;
+    }
+
+    public static int getNumberOfUsers()
+    {
+        return nrOfUsers;
+    }
+
+    public void setAge(final int age)
+    {
+        if (age < 18)
+        {
+            return;
+        }
+
+        this.age = age;
+    }
+
+    public void setName(final String name)
+    {
+        if (name.length() < 3)
+        {
+            return;
+        }
+
+        this.name = name;
+    }
+
+    public void setDateOfBirth(final LocalDate dateOfBirth)
+    {
+        this.dateOfBirth = dateOfBirth;
+    }
+
+    public void setNationality(final Nationality nationality)
+    {
+        this.nationality = nationality;
+    }
+
+    public void setHasJob(final boolean hasJob)
+    {
+        this.hasJob = hasJob;
+    }
+
+    @Override
+    public String toString()
+    {
+        return "The name of the user is: " + name
+                + "\nThe age of the user is: " + age
+                + "\nThe nationality of the user is: " + nationality
+                + "\nUser id is: " + idUser
+                + "\nUser is having a job: " + hasJob;
+    }
+
+    @Override
+    public boolean equals(Object object)
+    {
+        if (this == object)
+        {
+            return true;
+        }
+
+        if (object == null || getClass() != object.getClass())
+        {
+            return false;
+        }
+
+        User user = (User) object;
+
+        return age == user.age && Objects.equals(name, user.name)
+                && Objects.equals(nationality, user.nationality)
+                && Objects.equals(dateOfBirth, user.dateOfBirth);
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hashCode(name);
+    }
+
+    @Override
+    public int compareTo(User otherUser)
+    {
+        return otherUser.age - this.age;
+    }
+}
```

Comment:
Cateva lucruri extra:
 - deepcopy si shallow copy https://www.baeldung.com/java-deep-copy <- gasesti aici exemple la ambele cu explicatii ca stiu ca erai interesat
 - recomand sa incepi si un alt proiect. De exemplu un proiect de facut rezervari la o masa la orice restaurant. In prima faza, bazat pe ce am facut pana acum si la curs, ai putea face partea de class design, de structuri de date..etc.. Te poti gandi la ce fel de entitati o sa ai nevoie in aplicatia ta (de ex.: User, Restaurant, Table, Reservation, UserRoles... ce mai consideri tu), sa te gandesti la ce proprietati ai nevoie la fiecare si ce relatii pot exista intre ele.. de exemplu : un restaurant are un administrator (pe care il poti identifica cu un userId), un reservation trebuie sa aiba un user care a facut rezervarea (clientul) si unul care a aprobat (administratorul restaurantului)..etc.. pe viitor cand trecem la web poti chiar sa o dezvolti intr-o aplicatie web..acum ar fi suficient sa faci structura de clase si sa testez intr-o metoda main apeland metodele respective ca intr-un scenariu real.
 Cand te gandesti la entitati, recomand sa iti faci o schema cu relatii intre entitati ca sa ai un overview inainte sa te apuci.